core/domain/wipeout_service.py

```diff
@@ -476,10 +493,41 @@ def delete_user(
             ['manager_ids'])
         _pseudonymize_blog_post_models(pending_deletion_request)
         _pseudonymize_version_history_models(pending_deletion_request)
+        # Remove profile picture.
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
```

Comment:
Regarding this, I was not able to use the `get` here as I was getting `EntityNotFoundError` but when I use `get_by_id` it works fine. I tried to search if we somewhere use `get` with `UserSettingsModel`, I got no results but we do use `get_by_id` everywhere. I think this will be fine.
Thanks

Reply:
Wait that is weird, it should work almost the same as `get_by_id`. Can you please explain the issue more.

Reply:
I agree, I'm not sure what is going wrong here. When I use `get` instead of `get_by_id` I get the following error
```
Traceback (most recent call last):
  File "/home/lkbhitesh07/Desktop/oppia/oppia/core/domain/wipeout_service_test.py", line 5759, in test_log_error_when_profile_pictures_are_missing_while_deletion
    wipeout_service.delete_user(
  File "/home/lkbhitesh07/Desktop/oppia/oppia/core/domain/wipeout_service.py", line 507, in delete_user
    user_settings_model = user_models.UserSettingsModel.get(user_id)
  File "/home/lkbhitesh07/Desktop/oppia/oppia/core/storage/base_model/gae_models.py", line 358, in get
    raise cls.EntityNotFoundError(
core.storage.base_model.gae_models.BaseModel.EntityNotFoundError: Entity for class UserSettingsModel with id uid_ipuuzrdbeobooegbdewbbxqiglyyrxvs not found
```
After this, I thought I should take a look at other tests where we use `UserSettingsModel.get(user_id)` but there were none and when I checked for `UserSettingsModel.get_by_id(user_id)` there were several. Below you can find the relevant files:
[wipeout_service_test](https://github.com/oppia/oppia/blob/cf506ed05312ae3beaf079971b67e0a2061f81fa/core/domain/wipeout_service_test.py#L5719)
[wipeout_service](https://github.com/oppia/oppia/blob/cf506ed05312ae3beaf079971b67e0a2061f81fa/core/domain/wipeout_service.py#L410)

Reply:
Aha, I know what is going on, the user is marked as deleted and `get` takes that into account and only returns models that are not marked as deleted. Okay, I think it is fine to keep using `get_by_id` here.